src/hotspot/cpu/x86/stubGenerator_x86_64_cbrt.cpp

```diff
@@ -0,0 +1,366 @@
+    2790625147, 1016024251, 1330460186, 1016940346, 4033568463, 1015538390,
+    3695818227, 1017509621, 257573361,  1017208868, 3227697852, 1017337964,
+    234118548,  1017169577, 4009025803, 1017278524, 1948343394, 1017749310,
+    678398162,  1018144239, 3083864863, 1016669086, 2415453452, 1017890370,
+    175467344,  1017330033, 3197359580, 1010339928, 2071276951, 1015941358,
+    268372543,  1016737773, 938132959,  1017389108, 1816750559, 1017337448,
+    4119203749, 1017152174, 2578653878, 1013108497, 2470331096, 1014678606,
+    123855735,  1016553320, 1265650889, 1014782687, 3414398172, 1017182638,
+    1040773369, 1016158401, 3483628886, 1016886550, 4140499405, 1016191425,
+    3893477850, 1016964495, 3935319771, 1009634717, 2978982660, 1015027112,
+    2452709923, 1017990229, 3190365712, 1015835149, 4237588139, 1015832925,
+    2610678389, 1017962711, 2127316774, 1017405770, 824267502,  1017959463,
+    2165924042, 1017912225, 2774007076, 1013257418, 4123916326, 1017582284,
+    1976417958, 1016959909, 4092806412, 1017711279, 119251817,  1015363631,
+    3475418768, 1017675415, 1972580503, 1015470684, 815541017,  1017517969,
+    2429917451, 1017397776, 4062888482, 1016749897, 68284153,   1017925678,
+    2207779246, 1016320298, 1183466520, 1017408657, 143326427,  1017060403
+};
+
+#define __ _masm->
+
+address StubGenerator::generate_libmCbrt() {
+  StubGenStubId stub_id = StubGenStubId::dcbrt_id;
+  StubCodeMark mark(this, stub_id);
+  address start = __ pc();
+
+  Label L_2TAG_PACKET_0_0_1, L_2TAG_PACKET_1_0_1, L_2TAG_PACKET_2_0_1, L_2TAG_PACKET_3_0_1;
+  Label L_2TAG_PACKET_4_0_1, L_2TAG_PACKET_5_0_1, L_2TAG_PACKET_6_0_1;
+  Label B1_1, B1_2, B1_4;
+
+  address SIG_MASK        = (address)_SIG_MASK;
+  address EXP_MASK        = (address)_EXP_MASK;
+  address EXP_MSK2        = (address)_EXP_MSK2;
+  address EXP_MSK3        = (address)_EXP_MSK3;
+  address SCALE63         = (address)_SCALE63;
+  address ZERON           = (address)_ZERON;
+  address INF             = (address)_INF;
+  address NEG_INF         = (address)_NEG_INF;
+  address coeff_table     = (address)_coeff_table;
+  address rcp_table       = (address)_rcp_table;
+  address cbrt_table      = (address)_cbrt_table;
+  address D_table         = (address)_D_table;
+
+  __ enter(); // required for proper stackwalking of RuntimeStub frame
+
+  __ bind(B1_1);
+  __ subq(rsp, 24);
+  __ movsd(Address(rsp), xmm0);
+
+  __ bind(B1_2);
+  __ movq(xmm7, xmm0);
+  __ movl(rdx, 524032);
+  __ movsd(xmm5, ExternalAddress(EXP_MSK3), r11 /*rscratch*/);
+  __ movsd(xmm3, ExternalAddress(EXP_MSK2), r11 /*rscratch*/);
+  __ psrlq(xmm7, 44);
+  __ pextrw(rcx, xmm7, 0);
+  __ movdl(rax, xmm7);
+  __ movsd(xmm1, ExternalAddress(EXP_MASK), r11 /*rscratch*/);
+  __ movsd(xmm2, ExternalAddress(SIG_MASK), r11 /*rscratch*/);
+  __ andl(rcx, 248);
+  __ lea(r8, ExternalAddress(rcp_table));
+  __ movsd(xmm4, Address(r8, rcx, Address::times_1));
```

Comment:
This address and other instructions using similar address could be written as  Address(rcx, r8, Address::times_1).